src/GogeDao.sol

```diff
@@ -1,1087 +1,793 @@
-        SetTeamEnabled memory setTeamEnabled;
-        setTeamEnabled.description = poll.description;
-        setTeamEnabled.startTime = poll.time1;
-        setTeamEnabled.endTime = poll.time2;
-        setTeamEnabled.boolVar = poll.boolVar;
-
-        return (historicalTally[_pollNum], setTeamEnabled, passed[_pollNum]);  
+    /// @notice A method for removing polls from an address's advocatesFor mapped array.
+    /// @param _advocate address of wallet that we are removing their advocacy.
+    /// @param _pollNum the number of the poll the address is no longer an advocate for.
+    function _removeAdvocate(address _advocate, uint256 _pollNum) internal {
+        uint256 l = advocateFor[_advocate].length;
+        for (uint256 i; i < l;) {
+            if (advocateFor[_advocate][i] == _pollNum) {
+                advocateFor[_advocate][i] = advocateFor[_advocate][--l];
+                advocateFor[_advocate].pop();
+            }
+            unchecked {
+                i = i + 1;
+            }
+        }
     }
 
-    function getUpdateCakeDividendTracker(uint256 _pollNum) public view returns(uint256, UpdateCakeDividendTracker memory, bool) {
-        require(pollTypes[_pollNum] == PollType.updateCakeDividendTracker, "Not updateCakeDividendTracker");
-        Metadata memory poll = pollMap[_pollNum];
-        UpdateCakeDividendTracker memory updateCakeDividendTracker;
-        updateCakeDividendTracker.description = poll.description;
-        updateCakeDividendTracker.startTime = poll.time1;
-        updateCakeDividendTracker.endTime = poll.time2;
-        updateCakeDividendTracker.addr = payable(poll.addr1);
-
-        return (historicalTally[_pollNum], updateCakeDividendTracker, passed[_pollNum]);
+    /// @notice An internal method for changing gateKeeping status.
+    /// @param _enabled status of gateKeeping.
+    function _setGateKeeping(bool _enabled) internal {
+        require(gatekeeping != _enabled, "Already set");
+        gatekeeping = _enabled;
+        emit GateKeepingModified(_enabled);
     }
 
-    function getUpdateUniswapV2Router(uint256 _pollNum) public view returns(uint256, UpdateUniswapV2Router memory, bool) {
-        require(pollTypes[_pollNum] == PollType.updateUniswapV2Router, "Not updateUniswapV2Router");
-        Metadata memory poll = pollMap[_pollNum];
-        UpdateUniswapV2Router memory updateUniswapV2Router;
-        updateUniswapV2Router.description = poll.description;
-        updateUniswapV2Router.startTime = poll.time1;
-        updateUniswapV2Router.endTime = poll.time2;
-        updateUniswapV2Router.addr = payable(poll.addr1);
-
-        return (historicalTally[_pollNum], updateUniswapV2Router, passed[_pollNum]);
+    /// @notice An internal method for adding a team member address to the teamMembers array.
+    /// @param  _addr address of team member.
+    /// @param  _value is a team member.
+    function _setTeamMember(address _addr, bool _value) internal {
```

Comment:
First point. That's just preference I guess but when I think of "set" I think of setting a bool value to something inside a mapping. Update I think of changing or modifying a state variable like in _updateEndTime.
Second point. It's pretty much the same thing. Either way we're only calling isTeamMember() once. Maybe I can see which is more gas efficient but the current method makes sense in my head because in this function we are setting an address to be a specific boolean and first we need to see which boolean the user is trying to set the address to before moving any further.

Reply:
So after further testing I actually found that my method (check for value THEN isTeamMember) is cheaper than checking isTeamMember THEN value